### DISCUSSION TITLE
How to evaluate a string containing JavaScript code?

### DISCUSSION THREAD

**Discussion by itssamuelrowe** [webpack/webpack#15555] [Q&A]
Hello.

The following code evaluates and returns an object when executed with Node.js.

```js
const code = 'export default function hello() { console.log("Hello World"); }';
const dataUri =
  "data:text/javascript;charset=utf-8," + encodeURIComponent(code);
import(dataUri).then(console.log);
```

However, it fails when in React app, bootstrapped using Create React App. How can I solve this issue? Should I eject the application and modify the webpack configuration?

Thanks!

**Comment by vankop** [comment]
not sure what you are trying to achieve. you can import a file with `code` (like standard import). you can import asset as `asset/source` type https://webpack.js.org/guides/asset-modules/ (but this will not transpile/transform your code, e.g. `export` will not work).

If you trying to make fully "dynamic" module this will not work in webpack.

**Reply by itssamuelrowe** [reply]
Understood. Thank you for your response.

**Comment by ChocolateLoverRaj** [comment]
Did you try using a `Blob` with `URL.createObjectURL()`?
